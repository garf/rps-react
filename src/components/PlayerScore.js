import React, {Component} from 'react';
import PropTypes from 'prop-types';

class PlayerScore extends Component {
  render() {
    return (
      <div>{this.props.name}: {this.props.score}</div>
    );
  }
}

PlayerScore.propTypes = {
  name: PropTypes.string,
  score: PropTypes.number
};


export default PlayerScore;
