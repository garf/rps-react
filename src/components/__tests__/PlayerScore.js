import React from 'react';
import {shallow} from 'enzyme';
import PlayerScore from '@c/PlayerScore';

describe('PlayerScore component', () => {
  it('should render correct score for player', () => {
    const component = shallow(<PlayerScore name='Chuck Norris' score={3} />);

    expect(component).toMatchSnapshot();
  });
});
