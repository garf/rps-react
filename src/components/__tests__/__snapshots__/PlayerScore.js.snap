// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayerScore component should render correct score for player 1`] = `
<div>
  Chuck Norris
  : 
  3
</div>
`;
